Core/OfficeDevPnP.Core/Framework/Provisioning/ObjectHandlers/ObjectListInstance.cs

```diff
@@ -505,6 +515,33 @@ private void ProcessFieldRefs(Web web, Dictionary<Guid, Model.Field> siteFields,
             }
         }
 
+        private void ProcessPropertyBagEntries(TokenParser parser, PnPMonitoredScope scope, ListInfo list)
```

Comment:
```suggestion
        private void ProcessPropertyBagEntries(TokenParser parser, PnPMonitoredScope scope, ListInfo list)
{
	if (list.TemplateList.PropertyBagEntries != null && list.TemplateList.PropertyBagEntries.Count > 0)
	{
		// Handle root folder property bag
		var rootFolder = list.SiteList.RootFolder;
		list.SiteList.Context.Load(rootFolder, f => f.Properties);
		list.SiteList.Context.ExecuteQueryRetry();
		
		foreach (var p in list.TemplateList.PropertyBagEntries)
		{
			var parsedKey = parser.ParseString(p.Key);
			if (!rootFolder.Properties.FieldValues.ContainsKey(parsedKey) || p.Overwrite)
			{
				list.SiteList.SetPropertyBagValue(parsedKey, parser.ParseString(p.Value));
				if (p.Indexed)
				{
					list.SiteList.AddIndexedPropertyBagKey(parsedKey);
				}
				else
				{
					list.SiteList.RemoveIndexedPropertyBagKey(parsedKey);
				}
			}
			scope.LogDebug(CoreResources.Provisioning_ObjectHandlers_ListInstances_Added_PropertyBagEntry__0__To_List__1, p.Key, list.SiteList.Title);
		}
	}
}
```

1) Added token parsing support for keys 
2) Moved the logic to load rootfolder & properties inside the if condition so that we only them if the template contains the property bag values

Reply:
@patrikhellgren could you look into this feedback and resolve the merge conflicts?

Reply:
@gautamdsheth good suggestions so I committed your changes and also changed the LogDebug message to use parsedKey instead of p.Key.I used the Commit suggestion feature to commit your suggestions but then noticed that my code segment was still there. Not sure if I missed something with that feature or if you just added your new changes and left my code in place. Anyhow I sorted it out with another commit.

Reply:
@wobba I have now incorporated the suggested changes and resolved the merge conflicts.